db/migrate/20200331183640_add_repository_snapshots.rb

```diff
@@ -3,6 +3,7 @@
 class AddRepositorySnapshots < ActiveRecord::Migration[6.0]
   def up
     add_column :repositories, :parent_id, :bigint, null: true
+    add_column :repositories, :status, :integer, null: true
```

Comment:
Shouldn't this column have a default state? Especially if you are validating presence in the model.

Reply:
I think it should be explicitly set by business logic when snapshot is creating

Reply:
But we can set it to 'provisioning' by default

Reply:
Wait, this is STI, so better to have it nullable for simple repositories